build/xwalk.gni

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2016 Intel Corporation. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# This file contains Crosswalk-feature-related build flags. Users should
+# import it into their build folder by run:
+#   echo 'import("//xwalk/build/xwalk.gni")' > out/your_folder/args.gn'
+# Then run:
+#   gn gen out/your_folder
+
+# Set it false when building content_shell.
+declare_args() {
+  is_build_crosswalk = true
+}
+
+if (is_build_crosswalk) {
+  target_os = "linux"
+  target_cpu = "x64"
+  is_debug = false
+
+  # From src/build/config/features.gni
+  # TODO: Enable nacl for Linux.
+  enable_nacl = false
+
+  # From src/build/config/features.gni
+  enable_plugins = true
+
+  # From src/build/config/features.gni
+  enable_webvr = false
+
+  # From src/third_party/widevine/cdm/widevine.gni
+  enable_widevine = false
+
+  # From src/build/config/compiler/BUILD.gn
+  # If we want stack unwind support for backtrace()
+  exclude_unwind_tables = false
+
+  # From src/third_party/ffmpeg/ffmpeg_generated.gni
+  ffmpeg_branding = "Chromium"
+
+  # From src/third_party/icu/config.gni
+  icu_use_data_file = true
+
+  # From src/build/config/features.gni
+  proprietary_codecs = true
+
+  # From src/build/config/ui.gni
+  toolkit_views = true
+
+  # From src/build/config/ui.gni
+  use_aura = true
+
+  # From //build/config/sysroot.gni.
+  # Android and Windows do not use the sysroot, and we cannot use it on
+  # Linux because we use libnotify that is not present there.
+  use_sysroot = false
+
+  # From src/build_overrides/v8.gni
+  v8_use_external_startup_data = false
+
+  # From src/v8/BUILD.gn
+  v8_use_snapshot = false
+
+  # Crosswalk specific flags:
```

Comment:
I'd rather move the Crosswalk-specific options to a separate file, so that GN files that require access to these settings explicitly include it instead. It makes it a lot easier to understand what's overriding default upstream values and what are new variables.